libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/location/LocationPermissionComponent.kt

```diff
@@ -100,6 +67,21 @@ internal class LocationPermissionComponent(
         launcher?.unregister()
     }
 
+    private fun checkPermissions(applicationContext: Context) {
+        val isGranted = PermissionsManager.areLocationPermissionsGranted(applicationContext)
+
+        if (isGranted) {
+            onPermissionsResult(isGranted)
+        } else {
+            launcher?.launch(LOCATION_PERMISSIONS)
+            //  ActivityResultLauncher result is dispatched in ActivityResultCallback
```

Comment:
```suggestion
```

nitpicking: this comment does not help much

Reply:
The comment looks fine to me. Although I don't think this is necessary and rather add the type to the `launcher`

Change

```
private val launcher = try{ }
```

to

```
private val launcher: ActivityResultLauncher = try { }
```

Reply:
suggestion is to delete it because it is stating the obvious in my opinion. kind of qualifies for redundant constructs
https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

it stood out to me because of the placement. i never see comments describe the code above. if it's a good tl;dr for the [activity result launcher](https://developer.android.com/training/basics/intents/result), another suggestion is just reorder it
```
            //  ActivityResultLauncher result is dispatched in ActivityResultCallback
            launcher?.launch(LOCATION_PERMISSIONS)
```